src/components/CollapsibleTable.js

```diff
@@ -221,21 +124,67 @@ class CollapsibleTable extends Component {
   renderViewAction = (server) => {
     return (
       <span className="detail-info collapsible"
-        onClick={() => this.handleShowServerDetails(server)}>
+        onClick={() => this.setState({showServerDetailsModal: true, contextMenuRow: server})}>
         <i className="material-icons collapsible">info</i>
       </span>
     );
   }
 
-  renderMenuAction = (server) => {
+  getContextMenuItems = (row) => {
+    let items = [{
+      key: 'common.details', action: () => this.setState({showServerDetailsModal: true})
+    }];
+
+    if (row.role.includes('COMPUTE')) {
+      if (!isProduction()) {
+        // TODO: Add these as they are implemented
+        /*
+            key: 'common.activate', action: ...
+            key: 'common.deactivate', action: ...
+            key: 'common.delete', action: ...
+        */
+        // show replace button when there is no process operation going on
+        if (!this.props.processOperation) {
+          items.push({
+            key: 'common.replace',
+            action: this.props.replaceServer,
+            callbackData: row
+          });
+        }
+      }
+    } else {
+      // not compute node
+      if(!isProduction()) {
+        items.push({
```

Comment:
you want to have !this.props.processOperation as condition here as well

Reply:
ok, i'll add it. You didn't have it the version that you committed to master, so I figured you wanted it that way for some reason.

Reply:
Actually, let's add that in as a separate change, since this was not introduced by this PR